File: ObjC/Restaurant/Classes/Cells/FeedbackTableViewCell.m
//
//  FeedbackTableViewCell.m
//  Restaurants
//
//  Created by AppsFoundation on 1/24/15.
//  Copyright (c) 2015 AppsFoundation. All rights reserved.
//

#import "FeedbackTableViewCell.h"

@implementation FeedbackTableViewCell

- (void)awakeFromNib {
    // Initialization code
}

- (void)setSelected:(BOOL)selected animated:(BOOL)animated {
    [super setSelected:selected animated:animated];

    // Configure the view for the selected state
}

#pragma mark - Public

- (void)updateViewForRating:(int)rating {
    NSArray *arrayOfImages= @[self.oneStarImgView,self.twoStarsImgView,self.threeStarImgView,self.fourStarsImgView,self.fiveStarImgView];
    
    for (int i = 0; i < rating; i++) {
        UIImageView *imgView = arrayOfImages[i];
        imgView.hidden = NO;
    }
    
    for (int i = rating; i < arrayOfImages.count; i++) {
        UIImageView *imgView = arrayOfImages[i];
        imgView.hidden = YES;
    }
}

@end

File: Swift/Restaurant/Classes/Cells/FeedbackTableViewCell.swift
//
//  FeedbackTableViewCell.swift
//  Restaurant
//
//  Created by AppsFoundation on 8/27/15.
//  Copyright © 2015 AppsFoundation. All rights reserved.
//

import UIKit

class FeedbackTableViewCell: UITableViewCell {
    
    @IBOutlet weak var feedbackNameLabel: UILabel?
    @IBOutlet weak var feedbackTextLabel: UILabel?
    
    @IBOutlet weak var oneStarImgView: UIImageView?
    @IBOutlet weak var twoStarsImgView: UIImageView?
    @IBOutlet weak var threeStarImgView: UIImageView?
    @IBOutlet weak var fourStarsImgView: UIImageView?
    @IBOutlet weak var fiveStarImgView: UIImageView?
    
    override func awakeFromNib() {
        super.awakeFromNib()
    }
    
    override func setSelected(_ selected: Bool, animated: Bool) {
        super.setSelected(selected, animated: animated)
    }
    
    // MARK: - Public Methods
    func updateViewForRating(rating: Int) {
        let arrayOfImages: [UIImageView?] = [oneStarImgView, twoStarsImgView, threeStarImgView, fourStarsImgView, fiveStarImgView]
        
        for i in 0 ..< rating {
            let imgView = arrayOfImages[i]!
            imgView.isHidden = false
        }
        
        for i in rating ..< arrayOfImages.count {
            let imgView = arrayOfImages[i]!
            imgView.isHidden = true
        }
    }
}
